kolibri/plugins/coach/assets/src/views/plan/CoachExamsPage/index.vue

```diff
@@ -353,8 +385,25 @@
         }
         return '--';
       },
+      deviceContentUrl() {
+        const deviceContentUrl = urls['kolibri:kolibri.plugins.device:device_management'];
+        if (deviceContentUrl && this.canManageContent) {
+          return `${deviceContentUrl()}#/content`;
+        }
+
+        return '';
+      },
+      hasNoChannels() {
+        return !this.channels || this.channels.length === 0;
```

Comment:
[blocking, small cleanup] Doing this type of check is typically a good idea when fetching data, but it's unnecessary here. Since you have set up `channels` in `data` to be an empty array, `this.channels` will always exist, even if the channels haven't loaded yet. That means, the critical piece here is the "`this.channels.length === 0`"

[improvement, non-blocking] In your template code, you are using `channels.length === 0` on the banner and then you use this `hasNoChannels` computed prop for the button. I think it would be easier to see when reading the code that these are always "opposite" conditions, meaning that if one shows up, the other doesn't, if your code is more clearly referencing the same condition in both places. 

You could either update this computed prop to just be checking if `this.channels.length` and not worry about the existence check, and use `hasNoChannels` in both places, or you could delete this computed prop entirely and just use a length check in both places.